packages/sdk/typescript/human-protocol-sdk/test/escrow.test.ts

```
@@ -0,0 +1,1368 @@
+      ).rejects.toThrow();
+
+      expect(
+        escrowClient.escrowFactoryContract.createEscrow
+      ).toHaveBeenCalledWith(tokenAddress, trustedHandlers);
+    });
+  });
+
+  describe('setup', () => {
+    test('should throw an error if recordingOracle is an invalid address', async () => {
+      const escrowConfig = {
+        recordingOracle: FAKE_ADDRESS,
+        reputationOracle: ethers.constants.AddressZero,
+        recordingOracleFee: BigNumber.from(10),
+        reputationOracleFee: BigNumber.from(10),
+        manifestUrl: VALID_URL,
+        hash: FAKE_HASH,
+      };
+
+      await expect(
+        escrowClient.setup(ethers.constants.AddressZero, escrowConfig)
+      ).rejects.toThrow(ErrorInvalidRecordingOracleAddressProvided);
+    });
+
+    test('should throw an error if reputationOracle is an invalid address', async () => {
+      const escrowConfig = {
+        recordingOracle: ethers.constants.AddressZero,
+        reputationOracle: FAKE_ADDRESS,
+        recordingOracleFee: BigNumber.from(10),
+        reputationOracleFee: BigNumber.from(10),
+        manifestUrl: VALID_URL,
+        hash: FAKE_HASH,
+      };
+
+      await expect(
+        escrowClient.setup(ethers.constants.AddressZero, escrowConfig)
+      ).rejects.toThrow(ErrorInvalidReputationOracleAddressProvided);
+    });
+
+    test('should throw an error if reputationOracle is an invalid address', async () => {
+      const escrowConfig = {
+        recordingOracle: ethers.constants.AddressZero,
+        reputationOracle: ethers.constants.AddressZero,
+        recordingOracleFee: BigNumber.from(10),
+        reputationOracleFee: BigNumber.from(10),
+        manifestUrl: VALID_URL,
+        hash: FAKE_HASH,
+      };
+
+      await expect(
+        escrowClient.setup(FAKE_ADDRESS, escrowConfig)
+      ).rejects.toThrow(ErrorInvalidEscrowAddressProvided);
+    });
+
+    test('should throw an error if an hasEscrow return false', async () => {
```

fix writing:
'should throw an error if hasEscrow returns false'